src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -209,9 +214,11 @@ def _loop(self, session, options, options_bootstrapper):
       iterations -= 1
       while iterations and not self._state.is_terminating and not self._loop_condition.wait(timeout=1):
         continue
-    return target_roots, exit_code
+    return cast(TargetRoots, target_roots), exit_code
```

Comment:
Technically, this can be `None` so MyPy correctly infers it be `Optional`. But, in practice, it will always have a value and I didn't deem it worth refactoring (Pantsd code is not fun to mess around with).